pyperformance/data-files/benchmarks/bm_base64/run_benchmark.py

```diff
@@ -0,0 +1,274 @@
+DATA_LARGE = bytes(random.randrange(256) for _ in range(102400))
+DATA_HUGE = bytes(random.randrange(256) for _ in range(1048576))
+
+# Pre-encoded data for decode benchmarks
+B64_TINY = base64.b64encode(DATA_TINY)
+B64_SMALL = base64.b64encode(DATA_SMALL)
+B64_MEDIUM = base64.b64encode(DATA_MEDIUM)
+B64_9K = base64.b64encode(DATA_9K)
+B64_LARGE = base64.b64encode(DATA_LARGE)
+B64_HUGE = base64.b64encode(DATA_HUGE)
+
+B64_URLSAFE_TINY = base64.urlsafe_b64encode(DATA_TINY)
+B64_URLSAFE_SMALL = base64.urlsafe_b64encode(DATA_SMALL)
+B64_URLSAFE_MEDIUM = base64.urlsafe_b64encode(DATA_MEDIUM)
+B64_URLSAFE_9K = base64.urlsafe_b64encode(DATA_9K)
+
+B32_TINY = base64.b32encode(DATA_TINY)
+B32_SMALL = base64.b32encode(DATA_SMALL)
+B32_MEDIUM = base64.b32encode(DATA_MEDIUM)
+B32_9K = base64.b32encode(DATA_9K)
+B32_LARGE = base64.b32encode(DATA_LARGE)
+B32_HUGE = base64.b32encode(DATA_HUGE)
+
+B16_TINY = base64.b16encode(DATA_TINY)
+B16_SMALL = base64.b16encode(DATA_SMALL)
+B16_MEDIUM = base64.b16encode(DATA_MEDIUM)
+B16_9K = base64.b16encode(DATA_9K)
+B16_LARGE = base64.b16encode(DATA_LARGE)
+B16_HUGE = base64.b16encode(DATA_HUGE)
+
+A85_TINY = base64.a85encode(DATA_TINY)
+A85_SMALL = base64.a85encode(DATA_SMALL)
+A85_MEDIUM = base64.a85encode(DATA_MEDIUM)
+A85_9K = base64.a85encode(DATA_9K)
+A85_LARGE = base64.a85encode(DATA_LARGE)
+A85_HUGE = base64.a85encode(DATA_HUGE)
+
+B85_TINY = base64.b85encode(DATA_TINY)
+B85_SMALL = base64.b85encode(DATA_SMALL)
+B85_MEDIUM = base64.b85encode(DATA_MEDIUM)
+B85_9K = base64.b85encode(DATA_9K)
+B85_LARGE = base64.b85encode(DATA_LARGE)
+B85_HUGE = base64.b85encode(DATA_HUGE)
+
+
+# --- Base64 (includes validate=True) ---
+
+def bench_b64_small(loops):
+    range_it = range(loops)
+    t0 = pyperf.perf_counter()
+    for _ in range_it:
+        for _ in range(450):
+            base64.b64encode(DATA_TINY)
+            base64.b64decode(B64_TINY)
+            base64.b64decode(B64_TINY, validate=True)
+        for _ in range(71):
+            base64.b64encode(DATA_SMALL)
+            base64.b64decode(B64_SMALL)
+            base64.b64decode(B64_SMALL, validate=True)
+        for _ in range(3):
+            base64.b64encode(DATA_MEDIUM)
+            base64.b64decode(B64_MEDIUM)
+            base64.b64decode(B64_MEDIUM, validate=True)
+        base64.b64encode(DATA_9K)
+        base64.b64decode(B64_9K)
+        base64.b64decode(B64_9K, validate=True)
+    return pyperf.perf_counter() - t0
+
+
+def bench_b64_large(loops):
+    range_it = range(loops)
+    t0 = pyperf.perf_counter()
+    for _ in range_it:
+        for _ in range(10):
+            base64.b64encode(DATA_LARGE)
+            base64.b64decode(B64_LARGE)
+            base64.b64decode(B64_LARGE, validate=True)
+        base64.b64encode(DATA_HUGE)
+        base64.b64decode(B64_HUGE)
+        base64.b64decode(B64_HUGE, validate=True)
+    return pyperf.perf_counter() - t0
+
+
+# --- URL-safe Base64 (small only) ---
+
+def bench_urlsafe_b64_small(loops):
```

Comment:
They may be more rare, but there are more than 2 standard Base64 variants. For example, the variant with `altchars='+,'` is used for IMAP mailbox names (RFC 3501). It may be not worth to add benchmarks for them, but there would be a difference. `urlsafe_*()` are more optimized than `altchars=`.